code/Modules/Items/Carriables/ItemCleaner.cs

```diff
@@ -16,13 +19,41 @@ public ItemCleaner(ItemCleanerData itemData) : base(itemData)
         {
         }
         
+        /**
+         * The dirtyness is the level of sanity of the cleaner. 0 Is perfect and the max is to be defined.
+         */
+        public int Dirtyness { get; private set; }
+        
         public override void OnUsePrimary(MainPlayer player, ISelectable target)
         {
-            if (target is Stain stain)
+            switch (target)
             {
-                //Clean stain with the cleaning value
-                stain.CleanStain(Data.CleaningValue);
+                case Stain stain:
+                    Clean(stain);
+                    break;
+                case Bucket bucket:
+                    Wash(bucket);
+                    break;
             }
         }
+
+        /// <summary>
+        /// wash the cleaner in a bucket for set the dirty level to 0.
+        /// </summary>
+        /// <param name="bucket"></param>
+        private void Wash(Bucket bucket)
+        {
+            bucket.WasteWater(Dirtyness);
+            Dirtyness = 0;
+        }
+        
+        /**
+         * Clean stain with the cleaning value. The cleaning have to be decreased by the current dirtyness.
+         */
+        private void Clean(Stain stain)
+        {
+            stain.CleanStain(Data.CleaningValue);
+            Dirtyness++;
```

Comment:
Dirtyness += DirtFactor; ?
We could make the mop getting dirty quicker

Reply:
I want to keep it simple for now

Reply:
And there's no maximum for dirtyness '/?